wgpu-hal/src/gles/device.rs

```diff
@@ -412,11 +413,18 @@ impl crate::Device<super::Api> for super::Device {
             }
         }
 
+        let map_read_allocation = if emulate_map && desc.usage.contains(crate::BufferUses::MAP_READ) {
+            Some(std::sync::Mutex::new(vec![0; desc.size as usize]))
```

Comment:
It's a bit unfortunate to have this data buffer present in either `BufferInner::Data` or here. Can we refactor `struct Buffer` to avoid this duplication? Maybe
```
struct Buffer {
    raw: Option<glow::Buffer>,
    target: BindTarget,
    size: wgt::BufferAddress,
    map_flags: u32,
    map_data: Option<Mutex<..>>,
}
```

Reply:
good point, will fix